rules/group.go

```diff
@@ -507,6 +507,13 @@ func (g *Group) Eval(ctx context.Context, ts time.Time) {
 		ruleQueryOffset = g.QueryOffset()
 	)
 	eval := func(i int, rule Rule, cleanup func()) {
+		// Check if the group has been stopped.
+		select {
+		case <-g.done:
+			return
+		default:
+		}
```

Comment:
This should go outside of this loop or we'll execute:

```
g.metrics.GroupSamples.WithLabelValues(GroupKey(g.File(), g.Name())).Set(samplesTotal.Load())
g.cleanupStaleSeries(ctx, ts)
```

With incorrect metrics given that no queries were executed as the group as stopped.

Reply:
Done!